# Resolution of basal wolf-like canid divergenc eusing ancient DNA and fossil data

This project is an abridged version of a future, more involved project looking to resolve uncertain details in the wolf-like canid phylogeny through combined-evidence analysis using DNA, fossil, and morphological data under a Bayesian framework. The description below is currently intended to be for the simpler version of this project intended as a final project for EEOB563 - Molecular Phylogenetics, a graduate class at Iowa State University. More details, be it in data filtering, model building, model selection and others, will be considered in the future. 

## Background and question
The sequencing of ancient DNA for the dire wolf (*Canis dirus*) in recent work allowed for the analysis of a nuclear DNA phylogenetic tree that could elucidate debates related to the placement of dire wolves on the wolf-like canid (Canina) phylogeny (Perri et al 2021). While morphologically similar to the grey wolf (*Canis lupus*), the dire wolf does not appear closely related to it in a phylogenetic tree built from said data. Members of other genera (*Lycaon* and *Cuon*) of Canina appear to be closer to the grey wolf than *dirus*. This supports previous hypotheses that the dire wolf might be better placed on its own genus, *Aenocyon*. A similar debate has been carried for the African jackals, *Canis adustus* and *Canis mesomelas*. The phylogeny found in the 2021 study also supported the jackals' place as non-*Canis*, having them diverging from the rest of the Canina before the *Lycaon* and *Cuon* species as well.

The nuclear data in the paper failed, however, to reconcile the relationship between *dirus*, the African jackals and the rest of Canina. The question of which taxa, *dirus* or the jackals, diverged first from Canina can assist in future studies of morphological evolution of canids and provide more support to the place of *dirus* as a separate genus, which also encourages interesting work on its instance of convergent evolution. I plan to use combined-evidence analysis of DNA, fossil, and morphological data in a Bayesian framework to estimate a dated phylogenetic tree that can resolve the uncertainty on the order of divergence of basal Canina.

## Data
### Nuclear DNA
Nuclear DNA data was obtained from the authors of Perri et al 2021. It consists of an alignment with 19 samples of ~12 million SNPs consisting of 5 specimens of dire wolf, 3 specimens of grey wolf, 2 specimens of African wild dog (*Lycaon pictus*), and one each of the other representatives of the Canina subtribe, plus two outgroups. To clean the data, I first eliminated 3 dire wolf samples due to heavy damage, and one outgroup due to its absence from the morphological data set. Then, I removed sites that either contained missing data, or differed between specimens of the same group (dire wolves, gray wolves, African dogs, and the *Canis aureus*-*Canis lupaster* group, since these used to be classified as the same species and I am missing morphological data for *lupaster*). This resulted in a data set with 569583 bases spanning 9 Canina species and one outgroup (*Urocyon cinereoargenteus*, a fox).

### Fossil samples
I downloaded fossil samples from the *Canis*, *Lycaon*, and *Cuon* taxa. Then I kept only those with unique (name, min age, max age) combinations, and cut samples for which there was no accepted species name (mna) so as to minimize genus misidentification. Finally, I restricted this data set to the species that are either extant or present in the morphological data set (see below), so as to simplify the model. This did lead to the *Lycaon* and *Cuon* genera being represented by only its extant species with fossil samples, though that is unlikely to lead to much bias. One of the extant species (*Canis simensis*) did not have any fossil samples in the PBDB. In total, we collected 119 fossil samples spanning 9 extant and 13 extinct species. Note that the FBD and other models incorporating fossil tips and sampled ancestors can contain fossil samples that are not in the morphological data set, if one wishes.

### Morphological data
A morphological alignment was obtained from a previous study analyzing character evolution in extant and extinct canids (Slater 2015). This data set contains the scoring of 123 characters for 132 canids plus an outgroup. After restricting this data set to the species present either in the DNA or fossil data, we are left with 23 species, including our 10 extant species plus 13 fossil species. I then set all characters with all equal values for our final 23 species to their maximum value in the original data frame due to a RevBayes quirk. I deleted a row with all 0's and a row with all 4's since that would mean we'd have a whole model for just one character each. Deleting the characters for which there is missing data for all the species in the data set, we are left with a total of 104 characters. Note that the morphological data contains duplicate taxa, since we must copy the morphology of species X for each instance of a fossil sample of X. This is simply due to the nature of RevBayes algorithms.

## Analysis
The current plan is to follow the outline of the specimen-based Fossilized-Birth-Death model tutorial in the [RevBayes website](https://revbayes.github.io/tutorials/fbd/fbd_specimen.html). While this is a good baseline, I will also incorporate some partitioning, variation in molecular evolution models and prior sensitivity analysis for diversification rates. It is not clear to me yet what the exact procedure will be since this will depend on how much I can get done before the deadline, but the focus is to search for a robust combination of priors and models that lead to low uncertainty in the relationship between *dirus*, the jackals and the rest of the Canina. To make things more modular, I might first estimate a preliminary ultrametric tree using only the DNA data to use as a tree prior. 

Other details, such as the analysis of singletons and their effects on the placement of *dirus* (see Perri et al 2021, SI), will be considered as time permits.

### Model averaging: Nuclear DNA
For the nuclear DNA data, I set up model averaging on RevBayes (see `scrips/avg_nuclear_MCMC.Rev`) by considering 5 options of exchange matrices (Jukes-Cantor, Kimura 80, Felsenstein 81, Hasegawa-Kishino-Yano, and General Time Reversible), 2 options of among-site rate variation (`alpha` being `1E8` to represent no ASRV, or being drawn from a very diffuse uniform distribution), and 2 options for site invariance (`pInv` being `0` to represent no site invariance, or bein drawn from a `Beta(1, 1)`). Rates of transition were set to 0 in GTR, since the data set only contains transversions due to the quirks of ancient DNA data. The analysis with `pInv` led to a bug, but that was not too bad since it ran for long enough to show with high ESS (~5000) that `pInv` should be 0 with high posterior probability. This was expected given this is a SNP data set. I then took `pInv` out of the analysis. Rerunning it, I reached a 95% confidence interval for `alpha` of `[1E7, 1E8]`, implying strongly that no ASRV is the highest posterior probability model. `Q_indicator`, the variable used to cycle through the different models of Q, never left 5 (representing the GTR model) after burnin, implying the likelihood of other models is low enough compared to GTR that those moves were never accepted. All parameters had ESS above 2000, with most having around 9000. We then leave this analysis with the following conclusion that  GTR, with no +G or +I, is the best model for molecular evolution in this data set.

### Model averaging: Morphological data
For the morphological data, I set up model averaging on RevBayes (see `scripts/avg_morpho_setup.Rev` and `scripts/avg_morpho_MCMC.Rev`) by considering 3 options of exchange matrices (Jukes-Cantor, Felsenstein 81, and General Time Reversible), 2 options of among-site rate variation (as above), and 2 options for partitioning of sites in categories (where we divided characters among 4 categories each with their own exchangeability rates and state frequencies). I ran a different analysis for each size of the sample space (i.e. for character with 2 states, 3 and 4), and for each I set `er_scale` and `pi_scale` parameters that could either be `62000` (in the case where all categories had the same rates) or come from a uniform distribution. Then these were used to get 4 category specific rates from a discretized beta with 4 categories and shape and scale equal to the corresponding parameter. These latter rates were finally used as scale/shape for another discretized beta with size equal to the corresponding state sample space size. I set up indicators for 1. which Q matrix we were at, 2. whether we had among-site rate variation (`alpha`), and 3. whether the rates should differ between categories (i.e. whether `er_scale` and `pi_scale` come from the 62000 or the uniform distribution). I found that: 1. For 2-state characters, the best model was a GTR+G model with among-category variation for both `er` and `pi`; 2. For 3-state characters, the best model was a simple JC model, and 3. For 4-state characters, 

## References
Perri, A.R., Mitchell, K.J., Mouton, A. et al. Dire wolves were the last of an ancient New World canid lineage. Nature 591, 87â€“91 (2021). https://doi.org/10.1038/s41586-020-03082-x.

Slater, G. J. Iterative adaptive radiations of fossil canids show no evidence for diversity-dependent trait evolution. PNAS 112(16), 4897-4902. https://doi.org/10.1073/pnas.1403666111.
